Modules/System/Extension Management/src/ExtensionManagement.Codeunit.al

```diff
@@ -82,6 +82,18 @@ codeunit 2504 "Extension Management"
         exit(ExtensionOperationImpl.DownloadExtensionSource(PackageId));
     end;
 
+    /// <summary>
+    /// Retrives the source of an extension, based on its PackageId.
+    /// </summary>
+    /// <param name="PackageId">The PackageId of the extension.</param>
+    /// <param name="TempBlob">TempBlob where the zip is stored.</param>
+    /// <param name="CleanFileName">A filename identifyning the extension</param>
+    /// <returns>True if the operation was successful; false otherwise.</returns>
+    procedure GetExtensionSource(PackageId: Guid; var TempBlob: Codeunit "Temp Blob"; var CleanFileName: Text): Boolean
```

Comment:
Why is the file name needed here?

For the downloading part, it's needed because that's the name of the file that's downloaded. But here, seems like an unneeded extra parameter.

Reply:
You are correct that the filename is not needed. However, I kept it because it contains information about the extension in a human readable format

Reply:
If you needed in your implementation, then add it as a separate function. In any case, the parameter should be removed from `GetExtensionSource`.

Reply:
Fine with me - you can just remove it

Reply:
Do you need more from me?

Reply:
No, we're done :-) Code is in. Ships in March!